tests/test_runs/test_run_functions.py

```diff
@@ -253,3 +255,48 @@ def test_get_runs_list_by_filters(self):
     def test_get_runs_list_by_tag(self):
         runs = openml.runs.list_runs(tag='curves')
         self.assertGreaterEqual(len(runs), 1)
+
+    def test_run_on_dataset_with_missing_labels(self):
+        from openml.runs.functions import _prediction_to_row
+        from sklearn.tree import DecisionTreeClassifier
+        from sklearn.preprocessing.imputation import Imputer
+        task = openml.tasks.get_task(2)
+        class_labels = task.class_labels
+
+        model = Pipeline(steps=[('Imputer', Imputer(strategy='median')),
+                                ('Estimator', DecisionTreeClassifier())])
+
+        X, Y = task.get_X_and_y()
+        rep_no = 0
+        # TODO use different iterator to only provide a single iterator (less
+        # methods, less maintenance, less confusion)
+        for rep in task.iterate_repeats():
```

Comment:
I'm not sure what this code is exactly doing here. In the end, you only want to test _prediction_to_row, right?